keps/sig-auth/20190916-noderestriction-pods.md

```diff
@@ -174,21 +171,26 @@ makes the following assumptions:
 
 ## Proposal
 
-### Node Restrictions
-
 All restrictions will be enforced through the [NodeRestriction][] admission controller. These
 extensions will be guarded by the `MirrorPodNodeRestriction` feature gate.
 
-On update, only the delta will be restricted. In other words, the Kubelet can modify pods with
-restricted labels / owner references as long as it's not modifying (or adding or
-deleting) the restricted entries.
+Nodes are not granted the `update` or `patch` permissions on pods, but may update `pod/status`. All
+label and owner reference updates will be forbidden through `pod/status` updates, so restrictions
+will not be checked on status updates. In other words, a node _can_ update the status on a pod that
+has un-whitelisted labels.
+
+### Label Restrictions
 
-#### Label Restrictions
+A new reserved annotation will be introduced for namespaces to whitelist mirror-pod labels:
 
-The Kubelet will be prevented from updating pod labels or creating mirror pods with labels, except
-for whitelisted keys:
+```
+node.kubernetes.io/mirror.allowed-label-keys = "key1,key2,..."
```

Comment:
nit: As opposed to serialized json array?  I generally prefer the stronger structure instead of splitting on commas.

Reply:
Is there a compelling benefit in this case? If the values are extremely constrained (as valid label keys are, which cannot contain whitespace, quotes, or commas), I prefer the simpler form. Since we can't easily validate the annotation at the point the user applies it to the namespace, requiring well-formed JSON is more likely to result in admission-time errors. 

For labels/selectors specifically, users are accustomed to expressing them as comma-separated strings. It's also quite a bit easier to see the intent and spot errors in `foo1` below as opposed to `foo2`:

```json
{
    "metadata": {
        "annotations": {
            "foo1": "bar,baz",
            "foo2": "[\"bar\",\"baz\"]"
        }
}
```